cartographer/cloud/internal/local_trajectory_uploader.cc

```diff
@@ -108,19 +107,15 @@ class LocalTrajectoryUploader : public LocalTrajectoryUploaderInterface {
 
 LocalTrajectoryUploader::LocalTrajectoryUploader(
     const std::string& uplink_server_address, int batch_size,
-    bool enable_ssl_encryption, const std::string& token_file_path)
+    bool enable_ssl_encryption, bool enable_google_auth)
     : batch_size_(batch_size) {
   auto channel_creds =
-      enable_ssl_encryption
-          ? ::grpc::SslCredentials(::grpc::SslCredentialsOptions())
-          : ::grpc::InsecureChannelCredentials();
-
-  if (!token_file_path.empty()) {
-    auto call_creds = async_grpc::TokenFileCredentials(
-        token_file_path, std::chrono::seconds(kTokenRefreshIntervalInSeconds));
-    channel_creds =
-        grpc::CompositeChannelCredentials(channel_creds, call_creds);
-  }
+      enable_google_auth
+          ? grpc::GoogleDefaultCredentials()
+          : enable_ssl_encryption
+                ? ::grpc::SslCredentials(::grpc::SslCredentialsOptions())
+                : ::grpc::InsecureChannelCredentials();
```

Comment:
Wow :) Can you put some brackets to clarify the evaluation order?

Reply:
Done. It's only one pair of parenthesis, but maybe it makes things easier to read :).